### PR TITLE
Add configurable timeout and improve shutdown resource release

### PR DESCRIPTION
**Pull request by ChuheZhang**
- **Configurable Timeout**: Added a `timeout` field to `Options` (default: 5 seconds), now used in `ServerOptions` for `http2_svr_ptr_`, allowing dynamic timeout adjustments.
- **Resource Release Optimization**: Updated the `kPostShutdown` hook to include `reset()` for `asio_executor_ptr_`, ensuring all resources are released properly.

### CODE REVIEW

File: src/plugins/grpc_plugin/grpc_plugin.cc [hunk 1]
```diff
@@ -113,8 +116,11 @@ bool GrpcPlugin::Initialize(runtime::core::AimRTCore* core_ptr) noexcept {
     asio_executor_ptr_->Start();
 
     core_ptr_->RegisterHookFunc(runtime::core::AimRTCore::State::kPostShutdown, [this] {
-      asio_executor_ptr_->Shutdown();
-      asio_executor_ptr_->Join();
+      if (asio_executor_ptr_) {  // Check if resource exists to ensure safe release
+        asio_executor_ptr_->Shutdown();
+        asio_executor_ptr_->Join();
+        asio_executor_ptr_.reset();  // Explicitly release the resource
```
Comment (zhangyi1357) [inline src/plugins/grpc_plugin/grpc_plugin.cc:122]: Defensive programming is good; let's add similar content to the 150-line range to maintain consistency.

**Review by zhangyi1357 — CHANGES_REQUESTED** [review]
Thank you for your hard work.

However, apart from some very minor changes, such as typos, please open an issue for discussion before adding any new features or fixing any bugs.  For example, we're completely lost regarding the timeout feature here; we have no idea what it's for.  Worse still, the related code can't even compile (you can run ./test.sh to compile and run unit tests).

**Comment by ChuheZhang** [comment]
Thank you for your feedback and guidance. I apologize for the oversight—I'm still new to some collaborative features in Git, including the protocol for discussing changes before implementing them. I'll make sure to open an issue and seek team input for any new features or bug fixes going forward. 

Regarding the timeout feature, I realize now that its purpose and implementation might not have been clear, and I’ll investigate the compilation issue using `./test.sh` as suggested. I appreciate your patience as I get up to speed with these collaborative processes, and I'll strive to follow the established workflow more closely.

**Comment by zhangyi1357** [comment]
> Thank you for your feedback and guidance. I apologize for the oversight—I'm still new to some collaborative features in Git, including the protocol for discussing changes before implementing them. I'll make sure to open an issue and seek team input for any new features or bug fixes going forward.
> 
> Regarding the timeout feature, I realize now that its purpose and implementation might not have been clear, and I’ll investigate the compilation issue using `./test.sh` as suggested. I appreciate your patience as I get up to speed with these collaborative processes, and I'll strive to follow the established workflow more closely.

You're welcome for any contributions. Don't feel afraid to ask for any help. I'm happy to provide guidance on any aspect, including familiarity with a particular part of the codebase, or git/GitHub collaboration workflows.

**Pull request closed without merging**